packages/kokkos/core/src/impl/Kokkos_Error.cpp

```diff
@@ -56,6 +56,28 @@
 
 namespace Kokkos {
 namespace Impl {
+static void (*violation_handler_pointer)(violation_info const&) = default_violation_handler;
+}
+
+void set_violation_handler(void (*new_handler)(violation_info const&)) {
+  violation_handler_pointer = new_handler;
+}
+
+namespace Impl {
+
+void default_violation_handler(violation_info const& info) {
+  fprintf(stderr, "contract violation in %s:%d\nfunction %s\n%s\n",
+      info.file_name, info.line, info.function_name, info.comment);
+  std::stringstream string_stream;
+  traceback_callstack(string_stream);
```

Comment:
Regarding `traceback_callback`'s [current implementation](https://github.com/kokkos/kokkos/blob/d3a941925cbfb71785d8ea68259123ed52d3f9da/core/src/impl/Kokkos_Error.cpp#L153), I had thought that `dlopen` only works for shared object files.  Not sure if it works for statically build executables etc.